app/lib/actions/pulp3/repository/save_artifact.rb

```diff
@@ -2,17 +2,18 @@ module Actions
   module Pulp3
     module Repository
       class SaveArtifact < Pulp3::Abstract
-        def plan(file, repository, tasks, unit_type_id)
-          file_name = file[:filename]
-          plan_self(:file_name => file_name, :repository_id => repository.id, :tasks => tasks, :unit_type_id => unit_type_id)
+        def plan(file, repository, tasks, unit_type_id, options = {})
+          options[:file_name] = file[:filename]
+          plan_self(:repository_id => repository.id, :tasks => tasks, :unit_type_id => unit_type_id, :options => options)
         end
 
         def run
           artifact_href = input[:tasks].last[:created_resources].first
           content_type = input[:unit_type_id]
           content_backend_service = SmartProxy.pulp_master.content_service(content_type)
-          data = content_backend_service.content_class.new(relative_path: input[:file_name], _artifact: artifact_href)
-          content_unit = content_backend_service.content_api.create(data)
+          upload_options = {_artifact: artifact_href}.merge(input[:options])
+          content = content_backend_service.create_content(upload_options.with_indifferent_access)
+          content_unit = content_backend_service.content_api.create(content)
           output[:content_unit_href] = content_unit._href
```

Comment:
I was thinking we use the options hash to add whatever needs to be added and on the content specific class we can check if all required options for that content_type are not nil. I am not sure what options will be needed for uploading other content types right now or whether those options will need to be passed to this action. Maybe we can look at that when supporting upload for different content types.

Reply:
yeah, i think thats fine for now